devito/propagator.py

```diff
@@ -83,11 +85,14 @@ def __init__(self, name, nt, shape, spc_border=0, time_order=0,
 
         # Cache blocking and default block sizes
         self.cache_blocking = cache_blocking
-        if(isinstance(block_size, Iterable)):
-            if(len(block_size) == len(shape)):
+        self.cache_block_inner = cb_inner_dim
+        if isinstance(block_size, Iterable):
+            if len(block_size) == len(shape):
                 self.block_sizes = block_size
+            elif len(block_size) == len(shape) - 1 and not self.cache_block_inner:
```

Comment:
That is one of the reasons why I added it. 
We still need a flag when we are looping, creating `for loops` with blocking, as at that time `len(block_size) == len(shape)` even if ignoring inner most dim:

 ```
  for spc_var, block_size in reversed(zip(list(self.space_dims), self.block_sizes)):
                 dim_var = str(self._var_map[spc_var])
                 loop_limits = self._space_loop_limits[spc_var]
-                old_upper_limit = loop_limits[1]
-                new_upper_limit = old_upper_limit-old_upper_limit % block_size
-                loop_limits = (new_upper_limit, old_upper_limit)
+
+                # Does not block inner most dim if cache_block_inner flag is set
+                if not inner_most_dim or self.cache_block_inner:
+                    loop_limits = (loop_limits[1] - (loop_limits[1] - loop_limits[0]) % block_size, loop_limits[1])
```

We can't skip the iteration even though we are not cb inner most dim in this case.

Furthermore when auto tuning `block_sizes` are ignored, thus the flag is still needed as seen above to differentiate which dimensions to block/prep for auto tuning.